src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1001,15 +1001,15 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         if (ASRUtils::is_integer(*left_type) && ASRUtils::is_integer(*right_type)) {
             bool is_l64 = ASR::down_cast<ASR::Integer_t>(left_type)->m_kind == 8;
             bool is_r64 = ASR::down_cast<ASR::Integer_t>(right_type)->m_kind == 8;
-            if (is_l64 != is_r64) {
+            if ((is_assign && is_l64 != is_r64) || (is_l64 && !is_r64)) {
```

Comment:
I think I wrote it incorrectly, sorry about that. This should be correct:
```suggestion
            if ((is_assign && (is_l64 != is_r64)) || (is_l64 && !is_r64)) {
```
I think the `&&` has precedence over `!=` otherwise, which we do not want.

Reply:
Okay, I'll fix that.